packages/extension-api/src/extension-api.d.ts

```diff
@@ -136,6 +136,37 @@ declare module '@podman-desktop/api' {
     readonly storagePath: string;
   }
 
+  /**
+   * A provider result represents the values a provider, like the {@linkcode ImageCheckerProvider},
+   * may return. For once this is the actual result type `T`, like `ImageChecks`, or a Promise that resolves
+   * to that type `T`. In addition, `null` and `undefined` can be returned - either directly or from a
+   * Promise.
+   *
+   * The snippets below are all valid implementations of the {@linkcode ImageCheckerProvider}:
+   *
+   * ```ts
+   * let a: ImageCheckerProvider = {
+   *  check(image: ImageInfo, token?: CancellationToken): ProviderResult<ImageChecks> {
+   *    return new ImageChecks();
+   *  }
+   *
+   * let b: ImageCheckerProvider = {
+   *  check(image: ImageInfo, token?: CancellationToken): ProviderResult<ImageChecks> {
+   * 		return new Promise(resolve => {
+   * 			resolve(new ImageChecks());
+   * 	 	});
+   * 	}
+   * }
```

Comment:
```suggestion
   * let b: ImageCheckerProvider = {
   *  async check(image: ImageInfo, token?: CancellationToken): ProviderResult<ImageChecks> {
   * 		return new ImageChecks();
   * 	}
   * }
```

use async instead of new Promise(...) 

Also we could do `return Promise.resolve()` but I think it's better to use async methods